src/common/mfu_flist_copy.c

```diff
@@ -1129,6 +1129,44 @@ static int mfu_create_file(mfu_flist list, uint64_t idx,
     return rc;
 }
 
+/* creates hardlink in destpath for specified file, identifies source path
+ * returns 0 on success and -1 on error */
+static int mfu_create_hardlink(mfu_flist list, uint64_t idx,
+        const mfu_param_path* srcpath,
+        const mfu_param_path* destpath, mfu_copy_opts_t* mfu_copy_opts)
+{
+    /* assume we'll succeed */
+    int rc = 0;
+
+    /* get source name */
+    const char* src_path = mfu_flist_file_get_name(list, idx);
+
+    /* get destination name */
+    const char* dest_path = mfu_param_path_copy_dest(src_path, 1,
+            srcpath, destpath, mfu_copy_opts);
+
+    /* No need to copy it */
+    if (dest_path == NULL) {
+        return 0;
+    }
+
+    rc = mfu_hardlink(src_path, dest_path);
+    if (rc != 0) {
+        MFU_LOG(MFU_LOG_ERR, "Failed to create hardlink %s --> %s\n",
+                dest_path, src_path);
+        mfu_free(&dest_path);
+        return -rc;
```

Comment:
Let's return the raw error rc here, rather than flipping its sign.  May need to update code that calls this correspondingly.

Reply:
Done